packages/maker.js/src/core/chain.ts

```diff
@@ -595,18 +627,32 @@ namespace MakerJs.chain {
                 if (link.reversed) {
                     r = 1 - r;
                 }
+                let chainPoint: IWalkChainPoints = {
+                    point: point.add(point.middle(wp.pathContext, r), wp.offset),
```

Comment:
My thought is that you'd want to know `r` in your scenario. This is the ratio of where the point sits on the path. If this were an Arc, you could use `r` to get the angle you might want for rotation.

Reply:
I wanted to capture the direction the path is taking and that could just be done with the end points for any type, line or arc to make it follow the flow of path.

Reply:
And I guess if rotation of the child needs to be customized it can be done now by just using the `toPoints `or `toKeyPoints ` depending upon the need of the use case since there can be scenarios where users of library might need to use bigger child elements to be placed on smaller paths and they can now customize it by checking `pathContext ` of the chain. Not sure if there any requests for this feature but seems like something that can be helpful.

Reply:
The title of the PR is "Rotation along path" so we will need to provide that angle the point of `r`. Alternatively, we might just provide `r` and let the consumer calculate the rotation.

Reply:
If we provide r "ratio" then again we will need to parse the whole chain (the computation that is already happening which will be wasted) to get to the path context which is actually helping me get the rotation and direction of the path at that point. If its only about the PR name that can be updated.

Reply:
I see, from [your comment](https://github.com/microsoft/maker.js/issues/561#issuecomment-1501746543) that you would like to use the endpoints of a chain link's path to get the angle you need. This might be a solution in your specific scenario, where you have "short" paths with only one key point on a path. I can imagine other scenarios where multiple key points are on a path. This is a case where we would need the `r` value. For example, if there are multiple key points along a single arc. With just a couple of minor changes we can accomplish both. I will add more comments on other code references.

Reply:
This object can just be the new interface, with just `link`, `keyPoint`, and `ratio`.

Reply:
Go it, the scenario with multiple key points on single path makes sense will update the PR.

Reply:
one thing for the other method `toKeyPoints` there is no ratio.

Reply:
do we need ratio? because we have the point on path and other path properties which can be used to get the ratio.

Reply:
See comments on the other method - let's back that out, because we don't have the ratio.

Reply:
Updated the structure of the response and also made ratio available for `toPoints` let mw know if this looks good or I should remove the ratio.